src/test/scala/is/hail/annotations/StagedRegionValueSuite.scala

```diff
@@ -0,0 +1,367 @@
+package is.hail.annotations
+
+import is.hail.SparkSuite
+import is.hail.asm4s.{Code, _}
+import is.hail.asm4s.Code._
+import is.hail.expr._
+import is.hail.utils._
+import org.testng.annotations.Test
+
+class StagedRegionValueSuite extends SparkSuite {
+
+  val showRVInfo = true
+
+  @Test
+  def testString() {
+    val rt = TString
+    val input = "hello"
+    val srvb = new StagedRegionValueBuilder[String](FunctionBuilder.functionBuilder[String, MemoryBuffer, Long], rt)
+
+    srvb.emit(srvb.start())
+    srvb.emit(srvb.addString(srvb.input))
+    srvb.build()
+
+    val region = MemoryBuffer()
+    val rv = RegionValue(region)
+    rv.setOffset(srvb.transform()(input, region))
```

Comment:
So my suggestion is basically to write this bit as:
```scala
val fb = FunctionBuilder.functionBuilder[String, MemoryBuffer, Long]
val srvb = new StagedRegionValueBuilder[String](fb, rt)
fb.emit(Code(
  srvb.start(),
  srvb.addString(fb.getArg[String](1)),
  Code._return(srvb.currentOffset)))
val f = fb.result()()

val region = MemoryBuffer()
val rv = RegionValue(region)
rv.setOffset(f(input,region))
```
It's a bit more verbose, but it makes clear that the FB accepts code, and the SRVB builds code.